salt/utils/vt.py

```diff
@@ -340,7 +340,7 @@ def _translate_newlines(self, data):
         if data is None or not data:
             return
         # PTY's always return \r\n as the line feeds
-        return data.replace('\r\n', os.linesep)
+        return data.decode('utf-8').replace('\r\n', os.linesep)
```

Comment:
We are only invoking `self._translate_newlines()` on strings which have already been passed through `salt.utils.stringutils.to_str()`. This means that when run on Python 3, `data` will be a `str` type, which has no `decode` attribute:

```zsh
% python3 -c 'print("foo".decode("utf-8"))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'str' object has no attribute 'decode'
```

A better fix would be to revert line 343, and change the calls to `salt.utils.stringutils.to_str()` (lines 645 & 678) to use `salt.utils.stringutils.to_unicode()` instead.

Reply:
Sorry I did not catch this earlier, I did not see it in the diff in my initial review.

Reply:
makes sense. doing it now